src/mount.c

```diff
@@ -104,10 +184,17 @@ pu_umount_all(const gchar *device,
 gboolean
 pu_device_mounted(const gchar *device)
 {
-    g_autofree gchar *cmd = g_strdup("mount");
-    g_autofree gchar *output = NULL;
+    g_autoptr(GList) mounted = NULL;
+
+    g_return_val_if_fail(g_strcmp0(device, "") > 0, FALSE);
+
+    if (!pu_find_mounted(device, &mounted)) {
+        g_printerr("Failed checking if device is mounted");
+        return TRUE;
+    }
 
-    g_spawn_command_line_sync(cmd, &output, NULL, NULL, NULL);
+    if (mounted)
+        return TRUE;
```

Comment:
Please reverse the logic of this if-statement. `if` should be used to check false or error states primarily. The function should proceed to the end in case everything is good or no errors occurred.